modules/models/behavior/motion_primitives/continuous_actions.hpp

```diff
@@ -19,7 +21,7 @@ class BehaviorMPContinuousActions : public BehaviorMotionPrimitives {
  public:
   BehaviorMPContinuousActions(const DynamicModelPtr& dynamic_model,
                              const commons::ParamsPtr& params)
-      : BehaviorMotionPrimitives(dynamic_model, params), motion_primitives_() {}
+      : BehaviorMotionPrimitives(params), motion_primitives_(), dynamic_model_(dynamic_model) {}
```

Comment:
Actually, the DynamicModel is also given for each agent!

agent.GetDynamicModel()

Have a look at BehaviorIDMClassic.